include/rviz_visual_tools/rviz_visual_tools.h

```diff
@@ -758,7 +758,7 @@ a   *        Warning: when using this in a loop be sure to call trigger() at end
    */
   bool publishAxis(const geometry_msgs::Pose& pose, scales scale = MEDIUM, const std::string& ns = "Axis");
   bool publishAxis(const Eigen::Affine3d& pose, scales scale = MEDIUM, const std::string& ns = "Axis");
-  bool publishAxis(const geometry_msgs::Pose& pose, double length = 0.1, double radius = 0.01,
+  bool publishAxis(const geometry_msgs::Pose& pose, double length, double radius = 0.01,
```

Comment:
this breaks API... if someone is using this package with just ``publishAxis(pose)`` it will not longer compile after this change. This cannot be changed until ROS-M

Reply:
this is a compiler error. If you only supply the first argument, the compiler doesn't know which function to use:

```
Starting  >>> rviz_visual_tools               

Errors     << rviz_visual_tools:make /home/andy/ros/ws_sandbox/logs/rviz_visual_tools/build.make.097.log
/home/andy/ros/ws_sandbox/src/rviz_visual_tools/src/rviz_visual_tools_demo.cpp: In constructor ‘rviz_visual_tools::Rvi\
zVisualToolsDemo::RvizVisualToolsDemo()’:
/home/andy/ros/ws_sandbox/src/rviz_visual_tools/src/rviz_visual_tools_demo.cpp:83:36: error: call of overloaded ‘publi\
shAxis(geometry_msgs::Pose&)’ is ambiguous
     visual_tools_->publishAxis(pose);
                                    ^
In file included from /home/andy/ros/ws_sandbox/src/rviz_visual_tools/src/rviz_visual_tools_demo.cpp:44:0:
/home/andy/ros/ws_sandbox/src/rviz_visual_tools/include/rviz_visual_tools/rviz_visual_tools.h:759:8: note: candidate: \
bool rviz_visual_tools::RvizVisualTools::publishAxis(const Pose&, rviz_visual_tools::scales, const string&)
   bool publishAxis(const geometry_msgs::Pose& pose, scales scale = MEDIUM, const std::string& ns = "Axis");
        ^
/home/andy/ros/ws_sandbox/src/rviz_visual_tools/include/rviz_visual_tools/rviz_visual_tools.h:761:8: note: candidate: \
bool rviz_visual_tools::RvizVisualTools::publishAxis(const Pose&, double, double, const string&)
   bool publishAxis(const geometry_msgs::Pose& pose, double length = 0.1, double radius = 0.01,
        ^
make[2]: *** [CMakeFiles/rviz_visual_tools_demo.dir/src/rviz_visual_tools_demo.cpp.o] Error 1
make[1]: *** [CMakeFiles/rviz_visual_tools_demo.dir/all] Error 2
make: *** [all] Error 2
cd /home/andy/ros/ws_sandbox/build/rviz_visual_tools; catkin build --get-env rviz_visual_tools | catkin env -si  /usr/\
bin/make --jobserver-fds=6,7 -j; cd -

Failed     << rviz_visual_tools:make           [ Exited with code 2 ]
Failed    <<< rviz_visual_tools                [ 13.1 seconds ]
[build] Summary: 0 of 1 packages succeeded.
[build]   Ignored:   47 packages were skipped or are blacklisted.
[build]   Warnings:  None.
[build]   Abandoned: None.
[build]   Failed:    1 packages failed.
[build] Runtime: 13.3 seconds total.

Compilation exited abnormally with code 1 at Mon Oct 16 08:31:29
```

Reply:
ah i see, good catch